astroquery/simbad/core.py

```diff
@@ -936,6 +1016,256 @@ def query_objectids_async(self, object_name, *, cache=True,
 
         return response
 
+    def list_tables(self, *, get_adql=False):
+        """The names and descriptions of the tables in SIMBAD.
+
+        Parameters
+        ----------
+        get_adql : bool, optional
+            Returns the ADQL string instead of querying SIMBAD.
+
+        Returns
+        -------
+        `~astropy.table.table.Table`
+        """
+        query = ("SELECT table_name, description"
+                 " FROM TAP_SCHEMA.tables"
+                 " WHERE schema_name = 'public'")
+        if get_adql:
+            return query
+        return self.query_tap(query)
+
+    def list_columns(self, *tables: str, keyword=None, get_adql=False):
+        """
+        Get the list of SIMBAD columns.
+
+        Add tables names to restrict to some tables. Call the function without
+        any parameter to get all columns names from all tables. The keyword argument
+        looks for columns in the selected Simbad tables that contain the
+        given keyword. The keyword search is not case-sensitive.
+
+        Parameters
+        ----------
+        *tables : str, optional
+            Add tables names as strings to restrict to these tables columns.
+        keyword : str, optional
+            A keyword to look for in column names, table names, or descriptions.
+        get_adql : bool, optional
```

Comment:
we may need to call this `get_query_payload` to be consistent with the rest of the astroquery API, though your name is more precise and makes sense. 

What do you think @keflavich, is it time to cleanup what `get_query_payload` does and return the sql/adql where it makes more sense than returning the GET/POST parameters?

Reply:
Thanks for bringing this, I forgot to raise my concerns about this argument.

I was also thinking about a `get_raw_votable` (or `get_votable`) option. The use cases I see are:
- for debug purposes when the votable does not parse correctly. This can occur because astropy's parser is more strict than the one of TOPCat that we use for our tests (see #2916)
- to use votable sugar (examples are MIVOT annotations that will appear with https://github.com/astropy/pyvo/pull/497, or a simple SAMP broadcast with `astropy.samp`, it works well, I tried.)

But I did not want to fall in the boolean trap cause these two arguments (`get_adql\get_query_payload` and `get_votable`) would be mutually exclusive.

Then my idea was to maybe deviate from the API in place and introduce an `output_format` argument that'd accept a `Literal(['astropy_table', 'adql_string', 'astropy_votable'])` and would default to `astropy_table`. 

What are your thoughts?

Reply:
> I was also thinking about a get_raw_votable (or get_votable) option.

Yes, those would be great and I vaguely recall that it came up multiple times in the past and there should be issues for it already either here or in pyvo.

> Then my idea was to maybe deviate from the API in place and introduce an output_format argument that'd accept ['astropy_table', 'adql_string', 'astropy_votable', 'raw_votable'] and would default to astropy_table. This could be extended too.
What are your thoughts?

lets factor out this discussion into a separate ticket and pull into the discussion other interested parties (I think Adrian will definitely be interested, maybe the ESA folks, too, even though they don't use pyvo (yet)). I might use a different arg name that makes is clearer that it's a debug option, and may not support all 4 options you listed, but overall it looks useful to have more tools for easy troubleshooting purposes.